spec/porky_lib/file_service_spec.rb

```diff
@@ -94,240 +94,301 @@ def write_test_file(data)
     tempfile
   end
 
-  it 'write encrypted data to S3' do
-    file_key = file_service.write(plaintext_data, bucket_name, default_key_id)
-    expect(file_key).not_to be_nil
-  end
+  describe '#write' do
+    it 'attempt to write with file nil raises FileServiceError' do
+      expect do
+        file_service.write(nil, bucket_name, default_key_id)
+      end.to raise_exception(PorkyLib::FileService::FileServiceError)
+    end
 
-  it 'write large encrypted data to S3' do
-    file_key = file_service.write(File.read("spec#{File::SEPARATOR}porky_lib#{File::SEPARATOR}data#{File::SEPARATOR}large_plaintext"),
-                                  bucket_name, default_key_id)
-    expect(file_key).not_to be_nil
-  end
+    it 'attempt to write with bucket name nil raises FileServiceError' do
```

Comment:
I know this has been like this but the test should read like a sentence:
describe + it
Example: 
#write raises FileServiceError when bucket name is nil